src/Microsoft.Framework.FileSystemGlobbing/Internal/PathSegments/LiteralPathSegment.cs

```diff
@@ -29,12 +33,31 @@ public override bool Equals(object obj)
             var other = obj as LiteralPathSegment;
 
             return other != null &&
-                string.Equals(other.Value, Value, StringComparison.Ordinal);
+                _comparisonType == other._comparisonType &&
+                string.Equals(other.Value, Value, _comparisonType);
         }
 
         public override int GetHashCode()
         {
-            return Value.GetHashCode();
+            switch (_comparisonType)
+            {
+                case StringComparison.CurrentCulture:
+                    return StringComparer.CurrentCulture.GetHashCode(Value);
+                case StringComparison.CurrentCultureIgnoreCase:
+                    return StringComparer.CurrentCultureIgnoreCase.GetHashCode(Value);
+                case StringComparison.Ordinal:
+                    return StringComparer.Ordinal.GetHashCode(Value);
+                case StringComparison.OrdinalIgnoreCase:
+                    return StringComparer.OrdinalIgnoreCase.GetHashCode(Value);
+#if !DNXCORE50
+                case StringComparison.InvariantCulture:
+                    return StringComparer.InvariantCulture.GetHashCode(Value);
+                case StringComparison.InvariantCultureIgnoreCase:
+                    return StringComparer.InvariantCultureIgnoreCase.GetHashCode(Value);
+#endif
+                default:
+                    throw new InvalidOperationException($"Unexpected StringComparison type: {_comparisonType}");
```

Comment:
I don't like this giant switch statement, but it's the so far the only way I find to get a `StringComparer` from a `StringComparison`. We need the former instance to get the hash code properly.

Another solution suggested by @Tratcher is to use `StringComparer` in other places. However it doesn't fly, since string compare methods like `string.BeginWiths` accepts `StringComparison` only. I didn't find a easy way to get `StringComparison` from `StringComparer` either.

If you have a better solution, please let me know.